File: backend/tiggle-root/tiggle/src/main/java/com/side/tiggle/domain/asset/model/Asset.java
package com.side.tiggle.domain.asset.model;

import lombok.*;

import javax.persistence.*;

@Entity
@Data
@EqualsAndHashCode(callSuper = false)
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@AllArgsConstructor
@Table(name = "assets")
public class Asset {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "name", nullable = false)
    private String name;

    @Column(name = "defaults", nullable = false)
    private boolean defaults;

    @Builder
    public Asset(String name, boolean defaults) {
        this.name = name;
        this.defaults = defaults;
    }
}

File: backend/tiggle-root/tiggle/src/main/kotlin/com/side/tiggle/domain/asset/model/Asset.kt
package com.side.tiggle.domain.asset.model

import javax.persistence.*

@Entity
@Table(name = "assets")
class Asset(
    @Column(name = "name", nullable = false)
    var name: String,

    @Column(name = "defaults", nullable = false)
    var defaults: Boolean
) {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null
}